From 293c065385ff091935aa90e19fd59ab0b24c7c1d Mon Sep 17 00:00:00 2001
From: sai surya teja <sssteja@habana.ai>
Date: Sun, 19 Mar 2023 13:15:31 +0200
Subject: [PATCH] [SW-93113] hl: pll: avoid unintentional integer overflow during multiply

This patch fixes an issue pointed by synopsis static analysis tool by
type casting one of the multiply arguments to u64 to avoid any potential
integer overflow errors.

"47778 Unintentional integer overflow
The expression's value may not be what the programmer intended, because
the expression is evaluated using a narrow (i.e. few bits) integer type.

In hl_pll_get_freq: An integer overflow occurs, with the result
converted to a wider integer type (CWE-190)"

Change-Id: I05454365e325906e4c62e30be49cf81a1a298905
---

diff --git a/lib/common/pll.c b/lib/common/pll.c
index f436375..f741a52 100644
--- a/lib/common/pll.c
+++ b/lib/common/pll.c
@@ -132,7 +132,7 @@
 		/* refdiv : bits 24-29 */
 		refdiv  = EXTRACT_BITS(reg, 24, 6);
 		out_freq = SYS_REF_CLOCK * fbdiv;
-		out_freq /= refdiv * (postdiv1 + postdiv_fix) * (postdiv2 + postdiv_fix);
+		out_freq /= ((u64)refdiv) * (postdiv1 + postdiv_fix) * (postdiv2 + postdiv_fix);
 	}
 
 	if (div_sel == HL_PLL_DIV_REF || div_sel == HL_PLL_DIV_PLL) {
